.buildkite/pipeline.yml

```diff
@@ -12,27 +12,54 @@
 # permissions and limitations under the License.
 
 steps:
-  - label: ':go: go mod download'
-    command: 'go mod download'
 
-  # We use a "wait" step here, because Go's module logic freaks out when
-  # multiple go builds are downloading to the same cache.
-  - wait
-
-  - label: 'build'
-    command: 'make'
+  - label: ":docker: Build"
+    env:
+      DOCKER_BUILDKIT: "1"
+    command:
+      >
+        docker build \
+          --progress=plain \
+          --file tools/docker/Dockerfile \
+          --target firecracker-containerd-test-naive \
+          --tag localhost/firecracker-containerd-test-naive:${BUILDKITE_BUILD_NUMBER} .
```

Comment:
Why "naive" here?  We're testing more than the naive snapshotter...

Reply:
The idea was that in this image we automatically start up the naive snapshotter; there could be other images we run e2e tests in that start up the devmapper snapshotter too.

I renamed this to `localhost/firecracker-containerd-e2etest-naive` and created another image called `localhost/firecracker-containerd-unittest` for the non-root tests, which might help clarify this a bit more?

Reply:
Are you planning to have a separate image that starts the dm snapshotter?  Otherwise it might be better to name it as "e2etest" and run both snapshotters inside.

Reply:
Yeah having separate images per snapshotter was the idea (have a `# TODO` towards the bottom of the Dockerfile mentioning this as I did not implement the devmapper image at this time).

I didn't consider that we could just start up both snapshotters in the same image, but all else being equal I like the idea of having separate images for separate snapshotter implementations better just to prevent unnecessary dependencies (i.e. if I just want to run a test against the naive snapshotter I shouldn't have to necessarily also build the devmapper one).

Reply:
Maybe we should have a better selector in our tests for the snapshotters then, since right now it'll run all of the tests for both snapshotters?

Reply:
Yeah I was conflating some things here. The tests that will actually need the snapshotter to startup automatically don't exist yet (it's the test for the shim exit that is pending all this). I was just running these other tests in the one that automatically starts up the snapshotter out of convenience for now because it also needs the various `--privileged` and other flags to run anyways.

Either way, what I did now was just use two images `firecracker-containerd-unittest` and `firecracker-containerd-unittest-nonroot`. These are the only images we use right now; neither starts either snapshotter up automatically.

I am leaving the later stages that do start up the snapshotter automatically because they provide some utility immediately (by enabling manual debugging) and are going to be used in the very near future for the shim exit test.